solr/core/src/java/org/apache/solr/handler/admin/ClusterStatus.java

```diff
@@ -89,14 +93,61 @@ public static Health combine(Collection<Health> states) {
     }
   }
 
-  public ClusterStatus(ZkStateReader zkStateReader, ZkNodeProps props) {
+  public ClusterStatus(ZkStateReader zkStateReader, SolrParams params) {
     this.zkStateReader = zkStateReader;
-    this.message = props;
-    collection = props.getStr(ZkStateReader.COLLECTION_PROP);
+    this.solrParams = params;
+    collection = params.get(ZkStateReader.COLLECTION_PROP);
```

Comment:
isn't it a little awkward that this param is initialized here despite lots of params being taken in getClusterStatus?   (change or not as you wish)

Reply:
other params except `collection` are used only in `getClusterStatus()`, but collection is also used in `fetchClusterStatusForCollOrAlias()`